### ISSUE TITLE
`devtools::check()` will miss package dependency if respective code only appears in function arguments

### ISSUE THREAD

**Issue by ramiromagno** [r-lib/devtools#2456]
Using `{devtools}` version 2.4.4..

Assume a vanilla package (no Imports whatsoever in `DESCRIPTION`) with only one source R file with one function only:

```R
#' Star wars dataset
#'
#' @param x A data frame.
#'
#' @export
star_wars <- function(x = dplyr::starwars) {
  print(x)
}
```

`devtools::check()` will be oblivious to the dependency on `{dplyr}`:

```
── R CMD check results ──────────────────────────────────────────────────────────────────────────── weird 0.1.0 ────
Duration: 7.9s

0 errors ✔ | 0 warnings ✔ | 0 notes ✔

R CMD check succeeded
```

If I try to include this dependency, then I get:

```
── R CMD check results ──────────────────────────────────────────────────────────────────────────── weird 0.1.0 ────
Duration: 7.8s

❯ checking dependencies in R code ... NOTE
  Namespace in Imports field not imported from: ‘dplyr’
    All declared Imports should be used.

0 errors ✔ | 0 warnings ✔ | 1 note ✖

```

**Comment by jennybc** [comment]
What you are noticing is behaviour, admittedly sort of unexpected and quirky, of `R CMD check`, not of `devtools::check()`. So it's not something we will or can fix.

There are ways to suppress this NOTE, which I've recorded here (and this was a good nudge for us to include this content in the R Packages book):

https://github.com/hadley/r-pkgs/issues/828

**Comment by ramiromagno** [comment]
Thanks Jenny!

**Comment by moodymudskipper** [comment]
FWIW I've reported it to r-devel and it's fixed in dev so should not be an issue from 4.3 onwards
